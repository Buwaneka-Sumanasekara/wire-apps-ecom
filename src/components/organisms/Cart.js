import React from 'react';
import { FlatList, View } from 'react-native';
import CartItem from '../molecules/CartItem';
import CartTotalSummary from '../atoms/CartTotalSummary';
import Spacer from '../atoms/Spacer';
import ListEmptyComponent from '../atoms/ListEmptyComponent';


const Cart = ({ cartItems, onChangeQty, onPressItem }) => {
    return (
        <View style={{ flex: 1 }}>
            <FlatList
                data={cartItems}
                extraData={cartItems}
                renderItem={({ item }) => <CartItem item={item}
                    onChangeQty={(item, size, qty) => onChangeQty(item, size, qty)}
                    onPressItem={(item) => onPressItem(item)}
                />}
                keyExtractor={(item) => item.uniqueId}
                ListFooterComponent={<Spacer isHeight={true} size={"[200px]"} />}
                showsHorizontalScrollIndicator={false}
                showsVerticalScrollIndicator={false}
                ListEmptyComponent={<ListEmptyComponent isLoading={false} text={"Cart is empty"} />}
            />
            <CartTotalSummary />
        </View>
    );

}

export default Cart;