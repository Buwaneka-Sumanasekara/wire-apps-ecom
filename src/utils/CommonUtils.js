
//Format price to 2 decimal places
export function formatPrice(price,currency="") {
   return `${price.toFixed(2)} ${currency}`;
}