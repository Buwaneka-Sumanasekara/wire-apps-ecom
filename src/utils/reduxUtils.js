

export const formatItemToCart = (product) => {
    return {
        id: product.id,
        name: product.name,
        price: product.price,
        quantity: 1,
        productInfo: product,
    }
}