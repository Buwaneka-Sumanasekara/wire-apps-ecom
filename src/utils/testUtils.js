import React from 'react'
import { render } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import { setupStore } from '../store/configureStore'


const initialStore = {
  cart: {
    items: [],
    totalAmount:0
  },

}

const store = setupStore(initialStore)

const customRender = (children) =>
  render(
    <Provider store={store}>
      {children}
    </Provider>,
  )


// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }