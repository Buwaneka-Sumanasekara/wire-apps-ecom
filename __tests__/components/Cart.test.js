import React from 'react';

import Cart from '../../src/components/organisms/Cart';
import cartItemJson from '../data/cartItem.json';
import { render } from '../../src/utils/testUtils'

describe('<Cart/>', () => {
    it('rendered sucessfully', () => {
        const {toJSON} = render(<Cart cartItems={[cartItemJson]} />)
        expect(toJSON()).toMatchSnapshot()
    });
});

