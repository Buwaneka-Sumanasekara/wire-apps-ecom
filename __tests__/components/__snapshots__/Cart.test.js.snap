// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart/> rendered sucessfully 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <ListEmptyComponent
        isLoading={false}
        text="Cart is empty"
      />
    }
    ListFooterComponent={
      <Spacer
        isHeight={true}
        size="[200px]"
      />
    }
    data={
      [
        {
          "currency": "GBP",
          "id": "4",
          "lineAmount": 50,
          "name": "PUMA Roma Basic",
          "price": 50,
          "productInfo": {
            "SKU": "1247",
            "brandName": "Puma",
            "colour": "purple",
            "description": "Complete that vintage 60’s look with the PUMA Roma Basic, a near carbon copy of the original.Features a leather, nubuck, or canvas upper for a range of style.EVA midsole adds cushioning.Rubber outsole supplies durable traction.",
            "id": "4",
            "mainImage": "https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/44abed599ef04cf5b728cab676d57b5d_35857601_fr_puma_sc7.jpeg",
            "name": "PUMA Roma Basic",
            "price": {
              "amount": "50.00",
              "currency": "GBP",
            },
            "qty": 1,
            "seletedSize": "8",
            "sizes": [
              "8",
              "9",
              "10",
              "11",
            ],
            "stockStatus": "IN STOCK",
          },
          "qty": 1,
          "size": "8",
          "uniqueId": "4-8",
        },
      ]
    }
    extraData={
      [
        {
          "currency": "GBP",
          "id": "4",
          "lineAmount": 50,
          "name": "PUMA Roma Basic",
          "price": 50,
          "productInfo": {
            "SKU": "1247",
            "brandName": "Puma",
            "colour": "purple",
            "description": "Complete that vintage 60’s look with the PUMA Roma Basic, a near carbon copy of the original.Features a leather, nubuck, or canvas upper for a range of style.EVA midsole adds cushioning.Rubber outsole supplies durable traction.",
            "id": "4",
            "mainImage": "https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/44abed599ef04cf5b728cab676d57b5d_35857601_fr_puma_sc7.jpeg",
            "name": "PUMA Roma Basic",
            "price": {
              "amount": "50.00",
              "currency": "GBP",
            },
            "qty": 1,
            "seletedSize": "8",
            "sizes": [
              "8",
              "9",
              "10",
              "11",
            ],
            "stockStatus": "IN STOCK",
          },
          "qty": 1,
          "size": "8",
          "uniqueId": "4-8",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
              {
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "flex-start",
                "flexBasis": "16.666667%",
                "opacity": 1,
                "paddingRight": 16,
              }
            }
          >
            <Image
              resizeMethod="resize"
              resizeMode="contain"
              source={
                {
                  "uri": "https://s3-eu-west-1.amazonaws.com/api.themeshplatform.com/media/44abed599ef04cf5b728cab676d57b5d_35857601_fr_puma_sc7.jpeg",
                }
              }
              style={
                [
                  {
                    "width": "100%",
                  },
                  {
                    "height": 48,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              [
                {
                  "flexBasis": "50%",
                },
                {
                  "alignItems": "flex-start",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "fontWeight": "700",
                  },
                  {
                    "marginBottom": 4,
                  },
                ]
              }
            >
              PUMA Roma Basic
            </Text>
            <View
              style={
                [
                  {
                    "display": "flex",
                  },
                  {
                    "flexDirection": "row",
                  },
                  {
                    "marginBottom": 4,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#64748b",
                    },
                  ]
                }
              >
                Price: 50.00 GBP
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "display": "flex",
                  },
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "backgroundColor": "#64748b",
                    },
                    {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    {
                      "color": "#fff",
                    },
                    {
                      "fontWeight": "700",
                    },
                    {
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                    },
                    {
                      "marginRight": 8,
                    },
                  ]
                }
              >
                Size: 8
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "flexBasis": "33.333333%",
                },
                {
                  "display": "flex",
                },
                {
                  "alignItems": "flex-end",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "color": "#000",
                  },
                  {
                    "fontWeight": "700",
                  },
                  {
                    "marginBottom": 8,
                  },
                ]
              }
            >
              50.00 GBP
            </Text>
            <View
              style={
                [
                  {
                    "display": "flex",
                  },
                  {
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#000",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  }
                }
              >
                <Text />
              </View>
              <TextInput
                editable={false}
                style={
                  [
                    {
                      "borderBottomWidth": 1,
                      "borderLeftWidth": 1,
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "borderBottomColor": "#d1d5db",
                      "borderLeftColor": "#d1d5db",
                      "borderRightColor": "#d1d5db",
                      "borderTopColor": "#d1d5db",
                    },
                    {
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                    },
                    {
                      "width": 40,
                    },
                    {
                      "textAlign": "center",
                    },
                    {
                      "alignItems": "center",
                    },
                  ]
                }
                value="1"
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#000",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  }
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "flexDirection": "column",
        },
        {
          "alignItems": "flex-end",
        },
        {
          "paddingBottom": 20,
          "paddingTop": 20,
        },
        {
          "paddingRight": 8,
        },
        {
          "backgroundColor": "#f1f5f9",
        },
        {
          "shadowColor": "rgba(0, 0, 0, 0.1)",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        },
        {
          "position": "absolute",
        },
        {
          "bottom": 0,
        },
        {
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontWeight": "400",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#475569",
            },
            {
              "marginLeft": 8,
            },
          ]
        }
      >
        No Of Items
      </Text>
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#475569",
            },
            {
              "marginLeft": 8,
            },
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontWeight": "400",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#1e293b",
            },
            {
              "marginLeft": 8,
            },
          ]
        }
      >
        Total
      </Text>
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#1e293b",
            },
            {
              "marginLeft": 8,
            },
          ]
        }
      >
        0.00 GBP
      </Text>
    </View>
  </View>
</View>
`;
